src/main/java/junitparams/internal/parameters/ParametersFromCustomProvider.java

```diff
@@ -28,11 +31,28 @@ public boolean isApplicable() {
     }
 
     private ParametersProvider instantiate(Class<? extends ParametersProvider> providerClass) {
-        try {
-            return providerClass.newInstance();
-        } catch (Exception e) {
-            throw new RuntimeException("Your Provider class must have a public no-arg constructor!", e);
+        Constructor<?>[] constructors = providerClass.getConstructors();
+        for (Constructor<?> constructor : constructors) {
```

Comment:
this can fail given 

```
MyClass(A, B) {}
MyClass(FrameworkMethod) {}
```

since if the ordering is such that the another non-zero argument constructor is first in the returned constructor array then the inner if statement  will fall through to invoking the n-arg constructor with no arguments - even if a valid constructor exists later in the array.

ideally this should rather than iterating over all constructors should fall-through until a matching constructor is found.

```java
try {
 Constructor <? extends ParametersProvider> cons = providerClass.getConstructor(FrameworkMethod.class);
 return cons.newInstance(frameworkMethod);
} catch (Exception ignored) {
} 
...

try {
 return providerClass.newInstance();
} catch (Exception e) {
  throw new RuntimeException("Your Provider class must have a public no-arg constructor!", e);
}
```

subsequently to also prevent needless lookup given a class with many test methods the construction should probably also be cached

see #109 - class ParametersProviderFactory